website/content/docs/secrets/kubernetes.mdx

```diff
@@ -0,0 +1,339 @@
+   namespace/test created
+   ```
+
+   Here is a simple set up of a service account, role, and role binding in the Kubernetes `test`
+   namespace with basic permissions we will use for this document:
+   ```yaml
+   apiVersion: v1
+   kind: ServiceAccount
+   metadata:
+     name: test-service-account-with-generated-token
+     namespace: test
+   ---
+   apiVersion: rbac.authorization.k8s.io/v1
+   kind: Role
+   metadata:
+     name: test-role-list-pods
+     namespace: test
+   rules:
+   - apiGroups: [""]
+     resources: ["pods"]
+     verbs: ["list"]
+   ---
+   apiVersion: rbac.authorization.k8s.io/v1
+   kind: RoleBinding
+   metadata:
+     name: test-role-abilities
+     namespace: test
+   roleRef:
+     apiGroup: rbac.authorization.k8s.io
+     kind: Role
+     name: test-role-list-pods
+   subjects:
+   - kind: ServiceAccount
+     name: test-service-account-with-generated-token
+     namespace: test
+   ```
+
+   We can create these objects with `kubectl apply -f`.
+
+3. Enable the Kubernetes Secrets Engine:
+
+   ```shell-session
+   $ vault secrets enable kubernetes
+   Success! Enabled the kubernetes Secrets Engine at: kubernetes/
+   ```
+
+   By default, the secrets engine will mount at the same name as the engine, i.e.,
+   `kubernetes/` here. This can be changed by passing the `-path` argument when enabling.
+
+4. Configure the mount point. An empty config is allowed.
+
+   ```shell-session
+   $ vault write -f kubernetes/config
+   ```
+
+   Configuration options are available as specified in the
+   [API docs](/api-docs/secret/kubernetes).
+
+5. We can now configure Kubernetes Secrets Engine to create a Vault role (**not** the same as a
+   Kubernetes role) that can generate service account tokens for the given service account:
+
+   ```shell-session
+   $ vault write kubernetes/roles/my-role \
+       allowed_kubernetes_namespaces="*" \
+       service_account_name="test-service-account-with-generated-token" \
+       token_default_ttl="10m"
+   ```
+
+## Generating Credentials
+
+After a user has authenticated to Vault and has sufficient permissions, a read request to the
+`creds` endpoint for the Vault role will generate and return a new service account token.
+
+```shell-session
+$ vault write kubernetes/creds/my-role \
+    kubernetes_namespace=test
+
+Key                        Value
+–--                        -----
+lease_id                   kubernetes/creds/my-role/31d771a6-...
+lease_duration             10m0s
+lease_renwable             false
+service_account_name       test-service-account-with-generated-token
+service_account_namespace  test
+service_account_token      eyJHbGci0iJSUzI1NiIsImtpZCI6ImlrUEE...
+```
+
+We can use the service account token above (`eyJHbG...`) with any Kubernetes API request that
+its service account is authorized for (through role bindings).
+
+```shell-session
+$ curl -sk $(kubectl config view -o 'jsonpath={.clusters[0].cluster.server}')/api/v1/namespaces/test/pods \
```

Comment:
Is it possible to avoid the hardcoded 0? It would be easy to miss for people who have multiple clusters set up in their kubeconfig

Reply:
I'm honestly not sure. This seemed to be the most straightforward way I could come up with to tell me "what is the current kubernetes API host to use".

Reply:
It's maybe too complex/ugly to actually use, and I couldn't find a way with just jsonpath, but I couldn't resist having a stab at this :)

```bash
kubectl config view -o json | jq -r '."current-context" as $current | .clusters[] | select(.name == $current).cluster.server'
```

Reply:
In theory, `-o jsonpath='{.clusters[?(@.name == $.current-context)]}'` should work as well, but for some reason, `kubectl` doesn't seem to support anchoring back to the root object with `$` inside a filter expression.

Reply:
Something like this may also work. The minify option removes everything that's not in the currently selected context IIRC: 
```
kubectl config view --minify -o jsonpath='{.clusters[].cluster.server}'
```

Reply:
Oh nice! That's way better

Reply:
Cool. I'll do `--minify`.

Just for fun, there's also:

```sh
curl -sk $(kubectl config view -o jsonpath="{.clusters[?(@.name == \"$(kubectl config current-context)\")].cluster.server}")/api/v1/namespaces/test/pods --header "Authorization: Bearer ..."
```